### ISSUE TITLE
Newly added gamepad has ID which is out of bounds of gamepad manager's internal array.

### ISSUE THREAD

**Issue by ecchilds** [gurkenlabs/litiengine#431]
**Describe the bug**
Index of a newly added gamepad was out of bounds. Was likely due to a failure to add other gamepads, which did not exist, but which were still picked up by the app. Specifically, a Huion Pentablet driver seemed to ping the gamepad manager, likely incrementing the default gamepad index without actually adding to the manager's gamepad array.

Notably, this does not crash the game, but it does make gamepads unusable.

**Stack Trace**
```java
Dec 23, 2021 6:15:19 PM de.gurkenlabs.litiengine.UpdateLoop update
SEVERE: Index 2 out of bounds for length 1
java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 1
	at java.base/java.util.concurrent.CopyOnWriteArrayList.elementAt(CopyOnWriteArrayList.java:385)
	at java.base/java.util.concurrent.CopyOnWriteArrayList.get(CopyOnWriteArrayList.java:398)
	at de.gurkenlabs.litiengine.input.GamepadManager.get(GamepadManager.java:164)
	at de.gurkenlabs.litiengine.input.GamepadEntityController.retrieveGamepadValues(GamepadEntityController.java:76)
	at de.gurkenlabs.litiengine.input.GamepadEntityController.update(GamepadEntityController.java:43)
	at de.gurkenlabs.litiengine.UpdateLoop.update(UpdateLoop.java:154)
	at de.gurkenlabs.litiengine.UpdateLoop.process(UpdateLoop.java:138)
	at de.gurkenlabs.litiengine.GameLoop.process(GameLoop.java:86)
	at de.gurkenlabs.litiengine.UpdateLoop.run(UpdateLoop.java:57)
```
Probably related error message:
```
Dec 23, 2021 6:15:00 PM net.java.games.input.ControllerEnvironment log
INFO: Failed to initialize device HUION HID because of: java.io.IOException: Failed to acquire device (8007001e)
```

**To Reproduce**
(Probable) steps to reproduce the behavior:
1. Install (very old) Huion H610pro pentablet drivers. 
2. Run application within IDE.
3. Connect Gamepad controller.

**Expected behavior**
Index of new Gamepad should be within bounds.

**Your System:**
 - OS: Windows
 - LITIENGINE version: 0.5.1-beta
 - Java JDK/JRE version: openjdk16.0.1
 - IDE: Intellij

**Additional context**
Notably, this still occurs even when the pentablet drivers are not running. It is possible the pentablet drivers are not a part of this issue, I am still not sure. I will update this thread if I get further information.

**Comment by Gamebuster19901** [comment]
This might have been resolved in c31fffdfc521682e0a31f68cdc28bed929a030e4, not sure though.

**Comment by Gamebuster19901** [comment]
Specifically, the issue was caused because `Input.gamepads().get()` was being called instead of `Input.gamepads().getByID()`
https://github.com/gurkenlabs/litiengine/blob/5fffe4f639e7840b16f9983aaf9b139effecee25/src/de/gurkenlabs/litiengine/input/GamepadEntityController.java#L92-L93

**Comment by ecchilds** [comment]
Ah, yeah, that does look like it would fix it. I guess I should update. Thanks for the help!

**Comment by Gamebuster19901** [comment]
I don't believe a version that has that commit and is buildable is available yet.

**Comment by ecchilds** [comment]
Welp, I ended up fixing it by implementing a near carbon copy of GamepadEntityController with the changes from that commit.

**Comment by nightm4re94** [comment]
We're planning to release v0.5.2 next week, thanks for standing by and merry Christmas!

**Comment by steffen-wilke** [comment]
Fixed in c31fffdfc521682e0a31f68cdc28bed929a030e4
